<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>768</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1280</width>
    <height>768</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1280</width>
    <height>768</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="glView" name="widget" native="true">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>21</y>
      <width>700</width>
      <height>700</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>700</width>
      <height>700</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>700</width>
      <height>700</height>
     </size>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>250</y>
      <width>241</width>
      <height>20</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="layout_points">
     <item>
      <widget class="QRadioButton" name="rb_no_points">
       <property name="font">
        <font>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>none</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="rb_circle_points">
       <property name="font">
        <font>
         <pointsize>13</pointsize>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>circle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="rb_square_points">
       <property name="font">
        <font>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>square</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_4">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>420</y>
      <width>161</width>
      <height>20</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="layout_lines">
     <item>
      <widget class="QRadioButton" name="rb_solid_lines">
       <property name="font">
        <font>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>solid</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="rb_dashed_lines">
       <property name="font">
        <font>
         <pointsize>13</pointsize>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>dashed</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_16">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>280</y>
      <width>211</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="layout_point_size">
     <item>
      <widget class="QLabel" name="label_point_size">
       <property name="text">
        <string>size:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="slider_point_size">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinbox_point_size">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_17">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>450</y>
      <width>211</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="layout_line_size">
     <item>
      <widget class="QLabel" name="label_line_size">
       <property name="text">
        <string>size:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="slider_line_size">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinbox_line_size">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>360</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>190</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>220</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Points</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>390</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Lines</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>130</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Background</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_point_color">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>320</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose color</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_line_color">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>490</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose color</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_bg_color">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>160</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose color</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="spinbox_model_scale">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>530</y>
      <width>57</width>
      <height>21</height>
     </rect>
    </property>
    <property name="decimals">
     <number>1</number>
    </property>
    <property name="minimum">
     <double>0.100000000000000</double>
    </property>
    <property name="maximum">
     <double>2.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QSlider" name="slider_move_z">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>440</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QDial" name="dial_rotate_y">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>340</y>
      <width>50</width>
      <height>50</height>
     </rect>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
    <property name="wrapping">
     <bool>false</bool>
    </property>
    <property name="notchesVisible">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>230</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="layout_x">
     <item>
      <widget class="QSpinBox" name="spinbox_move_x">
       <property name="minimum">
        <number>-100</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_x">
       <property name="text">
        <string>x</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinbox_rotate_x">
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_move">
    <property name="geometry">
     <rect>
      <x>1057</x>
      <y>200</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Move</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget_15">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>400</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="layout_z">
     <item>
      <widget class="QSpinBox" name="spinbox_move_z">
       <property name="minimum">
        <number>-100</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_z">
       <property name="text">
        <string>z</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinbox_rotate_z">
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QSlider" name="slider_move_y">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>350</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="slider_model_scale">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>530</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
    <property name="pageStep">
     <number>10</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="invertedControls">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>500</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Scale the model</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_rotate">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>200</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Rotate</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>40</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Projection</string>
    </property>
   </widget>
   <widget class="QSlider" name="slider_move_x">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>270</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-100</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget_14">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>310</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="layout_y">
     <item>
      <widget class="QSpinBox" name="spinbox_move_y">
       <property name="minimum">
        <number>-100</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_y">
       <property name="text">
        <string>y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinbox_rotate_y">
       <property name="minimum">
        <number>-360</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QDial" name="dial_rotate_x">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>260</y>
      <width>50</width>
      <height>50</height>
     </rect>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="wrapping">
     <bool>false</bool>
    </property>
    <property name="notchesVisible">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QDial" name="dial_rotate_z">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>430</y>
      <width>50</width>
      <height>50</height>
     </rect>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
    <property name="wrapping">
     <bool>false</bool>
    </property>
    <property name="notchesVisible">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>110</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="button_save_settings">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>590</y>
      <width>61</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Save</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>70</y>
      <width>160</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="layout_projection">
     <item>
      <widget class="QRadioButton" name="radioButton_ortho">
       <property name="text">
        <string>otrho</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton_frustum">
       <property name="text">
        <string>frustum</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="button_load_settings">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>590</y>
      <width>61</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Load</string>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>540</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>570</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Settings</string>
    </property>
   </widget>
   <widget class="Line" name="line_6">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>630</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="button_screenshot">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>660</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Screenshot</string>
    </property>
   </widget>
   <widget class="QPushButton" name="button_gif">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>660</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Gif</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_gif_timer">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>700</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="button_choose_file">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>90</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose file</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>glView</class>
   <extends>QWidget</extends>
   <header location="global">glview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
